patroni/ha.py

```diff
@@ -134,6 +137,9 @@ def load_cluster_from_dcs(self):
             self.old_cluster = cluster
         self.cluster = cluster
 
+        # Note time we last updated from DCS
+        self.cluster.dcs_last_seen = int(time.time())
+
```

Comment:
1. The `Cluster` object is inherited from a `namedtuple` and therefore immutable: https://github.com/zalando/patroni/blob/93efa91bbdc5a50cad4f99c4e99286340ab94ce8/patroni/dcs/__init__.py#L447-L461
2. Logically, the `Cluster` object contains only info that represents the state of Patroni cluster in DCS. I think the `dcs_last_seen` field doesn't fit there.

The AbstractDCS class is a better place for dcs_last_seen: https://github.com/zalando/patroni/blob/93efa91bbdc5a50cad4f99c4e99286340ab94ce8/patroni/dcs/__init__.py#L627-L656. It will also simplify changes in api.py: https://github.com/zalando/patroni/pull/2051/files#diff-13329cd3342d1757c2c9b4dd7a26c0170f4deee5d32eff5c2fd719cc9ff9375bR586-R588

Reply:
Thanks for the feedback Alexander, I'll look into it

Reply:
So you're saying there should be a new `_DCS_LAST_SEEN` = 'dcs_last_seen'` in there with all the boilerplate to it? AFAICT, that'd be a new REST API call like `https://localhost:8008/dcs_last_seen` am I getting this right?

Reply:
No. It is only about adding a new property to AbstractDCS. In the API if could be exposed like you suggested, but there will be no need to copy it's value (line 587)

Reply:
I've added a commit, can you tell whether this is more like what you had in mind?